README.md

```diff
@@ -0,0 +1,167 @@
+
+You may remember from our lesson on callbacks that one way to explain a callback function is to say that it encapsulates the logic about how our program should continue in the future. Keeping this in mind how should our program proceed once a get request to `/` is received? At the moment, the only code inside the callback function is a comment, directed at you dear learner, and a call to a method `end` on the `response` object. This is an important function. Let's see if you can modify the code inside in this callback to get the test for this route in our testing file (`test/server_test.js`) to pass.
+
+> In order to get oriented here, it might be a good idea to take a look at the node documentation for the `end` method on the `response` object. Also, if you look at the test, you'll notice that the test checks to see if the `Content-Type` "Headers", which signal to the browser what kind of data has been set appropriately. In this case, we are just sending some text so we can use the type "plain/text".
+
+When testing your code, you can of course run `npm test` to check if your code is passing the Hello World test passes. However, if you would like to do some manual testing, you can start the server by opening your terminal and running `node run-server.js`. This will start the server on port 3000. You can then test the server in your browser, by going to `http://localhost:3000`, or, better yet, test it by calling the browser using the curl command line tool, like so:
+
+```bash
+curl -X GET http://localhost:3000
+```
+
+> Note: We will be using the curl command line tool in the next section of this lab, so you may want to get familiar with it.
+
+## POSTing a Message
+
+Okay, so you got the server to respond: "Hello, World!" Good job! To a large extent, the code the you've just made work represents the main pattern that you'll use through this lab. In each case, you'll define a route, and then within a callback function that you pass in, you will define the "business logic" that the server should follow to produce the desired response. With that in mind, let's move on.
+
+Now that we've built our "Hello World" server, let's start to build out our messaging apparatus. The ability to add messages seems like a good place to start, no? Then we can work on our retrieval pathways.
+
+So what do we need to do? First, of all we need place to store the messages. Let's just store our messages in memory to keep things simple. This will mean that the messages we create won't persist if we shutdown the server, but that's okay for our purposes. To acheive this, let's a define a variable `messages` that will start as an empty list or Array. You can put this somewhere near the top of your `server.js` file:
+
+`let messages = [];`
+
+Note, that in the line above we are using the new ES6 variable assignment keyword `let`. You will need to put the compiler into strict mode for this to work, otherwise you'll likely get an error when you run the code.
```

Comment:
Could we add a parenthetical here, along the lines of, "(To put the compiler into strict mode, just add `"strict mode"` to the top of the file.)"